File: test-suite/python/security/jwt_verification_clean.py
import jwt
from jose import jwt as jose_jwt
from jwt import decode as decode_jwt


def decode_rs256(token: str, public_key: str):
    return jwt.decode(
        token,
        public_key,
        algorithms=["RS256"],
        audience="api",
        issuer="https://issuer.example.com/",
    )


def decode_with_expiration(token: str, key: str):
    return jwt.decode(token, key, algorithms=["HS256"], options={"verify_exp": True})


def jose_decode(token: str, key: str):
    return jose_jwt.decode(token, key, algorithms=["HS256"], audience="api")


def aliased_decode(token: str, key: str):
    return decode_jwt(token, key, algorithms=["HS256"], options={"verify_signature": True})

File: test-suite/js/security/jwt-verification-clean.ts
import jwt from "jsonwebtoken";
import { verify as verifyToken } from "jsonwebtoken";
import { jwtVerify } from "jose";

const ISSUER = "https://auth.example.com";
const AUDIENCE = "api";

export function verifyJsonwebtoken(token: string, publicKey: string): string | object {
  return jwt.verify(token, publicKey, {
    algorithms: ["RS256"],
    issuer: ISSUER,
    audience: AUDIENCE,
  });
}

export function verifyAliased(token: string, publicKey: string): string | object {
  return verifyToken(token, publicKey, {
    algorithms: ["HS256"],
    issuer: ISSUER,
    audience: AUDIENCE,
  });
}

export async function verifyJose(token: string, publicKey: CryptoKey): Promise<unknown> {
  const { payload } = await jwtVerify(token, publicKey, {
    issuer: ISSUER,
    audience: AUDIENCE,
  });
  return payload;
}
